cmd/ctr/overlaybd_conv.go

```diff
@@ -161,7 +210,18 @@ var convertCommand = cli.Command{
 			Name:   targetImage,
 			Target: newMfstDesc,
 		}
-		return createImage(ctx, cli.ImageService(), newImage)
+		err = createImage(ctx, cli.ImageService(), newImage)
+		if err != nil {
+			return err
+		}
+		if context.Bool("push-artifact") {
+			newImageManifest, err := images.Manifest(ctx, cs, newMfstDesc, platforms.Default())
+			if err != nil {
+				return err
+			}
+			return prepareArtifactAndPush(ctx, cs, srcImg.Target(), newImageManifest, targetImage, cli.ImageService())
```

Comment:
Can you check to make sure the srcImg isn't hardcoding a media type for the manifest descriptor, and is decoding it from the content_store